addons/website_sale_delivery/controllers/main.py

```diff
@@ -60,6 +60,11 @@ def cart_carrier_rate_shipment(self, carrier_id, **kw):
             res['error_message'] = rate['error_message']
         return res
 
+    @http.route(['/shop/get_payment_provider_carrier_compatibility'], type='json', auth='public', methods=['POST'], website=True)
+    def get_payment_provider_carrier_compatibility(self, **kw):
+        """Create a mapping between the id of the payment providers and a list of ids of the supported carriers"""
+        return {}
```

Comment:
Actually, I don't think that you need to make an RPC at all. Check how they implemented `website_sale_picking`, which is very similar to `website_delivery_ups`: the information on the "compatibility" (actually, simply whether the provider is of a given type) is stored as an attribute on the template and then read in JS.